src/compiler/error-detector.js

```diff
@@ -96,3 +94,27 @@ function checkExpression (exp: string, text: string, errors: Array<string>) {
     }
   }
 }
+
+function isValidExpression (exp: string) {
+  try {
+    new Function(`return ${exp}`)
+    return true
+  } catch (e) {
+    /* istanbul ignore next */
+    try {
+      const babylon = (new Function(`return require('babylon')`))()
+      /* $flow-disable-line */
+      babylon.parseExpression(exp, {
+        plugins: [
+          'objectRestSpread',
+          'functionBind',
+          'numericSeparator',
+          'optionalChaining'
+        ]
```

Comment:
@nickmessing Regarding the environment variable, I think it'd just be good to include any plugins that could possibly be needed in a template expression - which I think you've already done. I think it's better to allow _too much_ than too little here. If they're using a syntax that _really_ isn't supported by the user's own Babel config, they'll still see an error, just a syntax error as they would expect, rather than an invalid template error.